spec/controllers/v1/legislators_controller_spec.rb

```diff
@@ -22,38 +19,60 @@
     end
 
     context "good email" do
-      context "new user" do
-        before { get :index, { email: 'user@example.com' } }
+      let!(:campaign) { FactoryGirl.create(:campaign_with_reps, count: 3, priority: 1) }
+      let (:state)    { FactoryGirl.create(:state, abbrev: 'CA') }
+      let (:district) { FactoryGirl.create(:district, district: '13', state: state) }
+      let!(:senator)  { FactoryGirl.create(:senator, state: state, with_us: false) }
+      let!(:rep)      { FactoryGirl.create(:representative, district: district,
+                                                            with_us: true) }
+      let(:campaign_ids) { campaign.legislators.pluck(:id) }
+      let(:returned_ids) { assigns(:target_legislators).map{|tl| tl['id']} }
```

Comment:
Definitely helpful to hear your thoughts on this. Could you give me a link to calls_controller_spec?

I think the reason I went with integration-style testing in the controller is that I didn't have integration tests anywhere else. If I had integration tests somewhere else, I imagine I could just do functional testing and simplify these controller tests with mocks. Did you write integration tests for the calls API (in addition to the controller tests)?

Also, I know that's it's better to use `FactoryGirl.build` where possible, but most of what I'm testing relies on db queries to return results, so I didn't see away around actually creating records. Definitely open to suggestions if you see a better way.